pkg/engine/disconnected.go

```diff
@@ -0,0 +1,219 @@
+	"path/filepath"
+	"strings"
+
+	"github.com/containers/podman/v4/libpod/define"
+	"github.com/containers/podman/v4/pkg/bindings"
+	"github.com/containers/podman/v4/pkg/bindings/containers"
+	"k8s.io/klog/v2"
+)
+
+func extractZip(url, name string) error {
+	cache := "/opt/.cache/" + name + "/"
+	dest := cache + "HEAD"
+	trimDir := strings.TrimSuffix(url, path.Ext(url))
+	directory := filepath.Base(trimDir)
+	absPath, err := filepath.Abs(directory)
+
+	data, err := http.Get(url)
+	if err != nil {
+		if _, err := os.Stat(dest); err == nil {
+			// remove the diff file
+			err = os.Remove(dest)
+			if err != nil {
+				klog.Info("Failed to remove file ", dest)
+				return err
+			}
+		}
+		klog.Info("URL not present...requeuing")
+		return nil
+	} else if data.StatusCode == http.StatusOK {
+		if _, err := os.Stat(dest); os.IsNotExist(err) {
+			defer data.Body.Close()
+			// Check the http response code and if not present exit
+			klog.Infof("loading disconnected archive from %s", url)
+			// Place the data into the placeholder file
+
+			// Unzip the data from the http response
+			// Create the destination file
+			os.MkdirAll(directory, 0755)
+
+			outFile, err := os.Create(absPath + "/" + name + ".zip")
+			if err != nil {
+				klog.Error("Failed creating file ", absPath+"/"+name+".zip")
+				return err
+			}
+
+			// Write the body to file
+			io.Copy(outFile, data.Body)
+
+			// Unzip the file
+			r, err := zip.OpenReader(outFile.Name())
+			if err != nil {
+				klog.Infof("error opening zip file: %s", err)
+			}
+			for _, f := range r.File {
+				rc, err := f.Open()
+				if err != nil {
+					return err
+				}
+				defer rc.Close()
+
+				fpath := filepath.Join(directory, f.Name)
+				if f.FileInfo().IsDir() {
+					os.MkdirAll(fpath, f.Mode())
+				} else {
+					var fdir string
+					if lastIndex := strings.LastIndex(fpath, string(os.PathSeparator)); lastIndex > -1 {
+						fdir = fpath[:lastIndex]
+					}
+
+					os.MkdirAll(fdir, f.Mode())
+					f, err := os.OpenFile(
+						fpath, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, f.Mode())
+					if err != nil {
+						return err
+					}
+					defer f.Close()
+
+					_, err = io.Copy(f, rc)
+					if err != nil {
+						return err
+					}
+				}
+			}
+			err = os.Remove(outFile.Name())
+			if err != nil {
+				klog.Error("Failed removing file ", outFile.Name())
+				return err
+			}
+			createDiffFile(name)
+			return nil
+		} else {
+			klog.Info("No changes since last disonnected run...requeuing")
+		}
+	}
+	return nil
+}
+
+func localDevicePull(name, device, trimDir string, image bool) (id string, err error) {
+	// Need to use the filetransfer method to populate the directory from the localPath
+	ctx := context.Background()
+	conn, err := bindings.NewConnection(ctx, "unix://run/podman/podman.sock")
+	if err != nil {
+		klog.Error("Failed to create connection to podman")
+		return "", err
+	}
+	// Ensure that the device is present
+	_, exitCode, err := localDeviceCheck(name, device, "-"+trimDir)
```

Comment:
When you call this in the image method (pkg/engine/image.go:118), you pass "-" + trimDir as an arg, so is prepending "-" also necessary in this call to `localDeviceCheck`?